WIP:  PUBSTWO-1716 (Bump spring-boot-starter-parent from 2.1.9.RELEASE to 2…

orosz-usgs opened the pull request:
….2.1.RELEASE)

  Updated spring-boot-starter-parent to 2.2.2.RELEASE
  Replaced removed MediaType.APPLICATION_JSON_UTF8_VALUE with MediaType.APPLICATION_JSON_VALUE
  Updated message converters to have a default characterset of UTF-8 to handle spring 2.2 deprecation:
https://stackoverflow.com/questions/58525387/mockmvc-no-longer-handles-utf-8-characters-with-spring-boot-2-2-0-release
  Added field DEFAULT_CHARSET in PubsConstantsHelper

Commit:
PUBSTWO-1716 (Bump spring-boot-starter-parent from 2.1.9.RELEASE to 2.2.1.RELEASE)
  Updated spring-boot-starter-parent to 2.2.2.RELEASE
  Replaced removed MediaType.APPLICATION_JSON_UTF8_VALUE with MediaType.APPLICATION_JSON_VALUE
  Updated message converters to have a default characterset of UTF-8 to handle spring 2.2 deprecation:
https://stackoverflow.com/questions/58525387/mockmvc-no-longer-handles-utf-8-characters-with-spring-boot-2-2-0-release
  Added field DEFAULT_CHARSET in PubsConstantsHelper

Commit:
PUBSTWO-1716 (Bump spring-boot-starter-parent from 2.1.9.RELEASE to 2.2.1.RELEASE)
  Fixed tests DeletedPublicationMvcServiceIT and PwPublicationMvcServiceSinglePubIT

dsteinich commented on src/main/java/gov/usgs/cida/pubs/PubsConstantsHelper.java:
```diff
@@ -1,10 +1,14 @@
 public final class PubsConstantsHelper {
 
 	public static final String DEFAULT_ENCODING = "UTF-8";
+	public static final Charset DEFAULT_CHARSET = StandardCharsets.UTF_8;
```
I'm not a big fan of shadowing constants. It also appears that this is only used in one place?

dsteinich commented on src/main/java/gov/usgs/cida/pubs/springinit/SpringConfig.java:
```diff
@@ -71,7 +71,10 @@ public void configureMessageConverters(List<HttpMessageConverter<?>> converters)
 		Jackson2ObjectMapperBuilder builder = new Jackson2ObjectMapperBuilder();
 		builder.featuresToDisable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
-		return new MappingJackson2HttpMessageConverter(builder.build());
+		MappingJackson2HttpMessageConverter converter = new MappingJackson2HttpMessageConverter(builder.build());
```
Does this change make the MvcService.setHeaders() obsolete? If so, please remove the code from MvcService.
Either way, please create test(s) to verify that it works.

dsteinich commented on src/test/java/gov/usgs/cida/pubs/webservice/DeletedPublicationMvcServiceIT.java:
```diff
@@ -38,13 +39,20 @@
 		MvcResult rtn = mockMvc.perform(get("/publication/deleted").accept(MediaType.APPLICATION_JSON))
 				.andExpect(status().isOk())
-				.andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
+				.andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
```
I'm not sure that this matches the actual content type delivered by the real service. These "integration" tests may need more massaging to better reflect the real output of the service. This may also require adding UTF-8 data to the tests.

dsteinich commented on src/test/resources/testData/sipp/dissemination.xml:
```diff
@@ -3,48 +3,48 @@
   <Abstract>There is an increasing need to provide actionable science for water management planning and ecological drought preparedness. However, questions remain regarding the most effective and efficient methods for extending scientific knowledge and products into management action and decision-making. This study analyzed two unique cases of water management in the context of ecologically avail ...
   <BasisNumber>RB00H5X.2</BasisNumber>
   <CostCenter>Fort Collins Science Center</CostCenter>
-  <IPPANumber xsi:nil="true" />
```
This is the actual input for the test from the real SIPP process. Please revert to the original data.

dsteinich commented on src/test/java/gov/usgs/cida/pubs/webservice/DeletedPublicationMvcServiceIT.java:
```diff
@@ -38,13 +39,20 @@
+
 	private MockMvc mockMvc;
 
+	@Before
```
These tests used to take advantage of Spring Boot test capabilities. Please go back to that rather than using straight Spring to set up the mockMvc.

Commit:
PUBSTWO-1716 (Bump spring-boot-starter-parent from 2.1.9.RELEASE to 2.2.1.RELEASE)
 Code not complete, WIP for feeedback:
   rolled back change to PubsConstantsHelper & SpringConfig
   rolled back test to autowired MockMnv
   rolled back dissemination.xml

Test still failing
